hack/generate/cli-doc/gen-cli-doc.go

```diff
@@ -79,6 +94,8 @@ func recreateDocDir(rootCmd *cobra.Command, docPath string) {
 		return "../" + base
 	}
 
+	htmlFormatter(rootCmd)
```

Comment:
I'd suggest we add a post-processor for replacing characters rather than changing the Cobra.Command directly.

Reply:
What I notice is that `GenMarkdownTreeCustom()` eventually calls Cobra packages which are a part of different upstream. That is why I had to replace the text in `Long` itself. Kindly let me know if I have understood your comment on the code correctly. If not please let me know how I can make the changes.

Reply:
I was thinking that we know the output from this markdown generator. Instead of changing the `Long` field before feeding the `Command` to the generator, I was thinking if we can read the generated markdown and do some text substitutions there.

Reply:
TL;DR I don't think we should post-process markdown files;

@tengqm this tool is purely meant to generate the docs that show up on the website. We're using `GenMarkdownTreeCustom` from cobra which does not offer a way to process the document being written. I do not want to traverse the markdown tree because we are only replacing the `<` and `>` for the `Synopsis` section where the problem occurs. It is a lot easier to change the Commands before they are written to the markdown files than to parse markdown looking for the correct section to manipulate.

If cobra had a mechanism to pass in a custom writer for the markdown file or if there was a post-process handler that existed that we could configure during the generation process I would say yes let's post-process. 

Also like I mentioned before, this tool is purely for generating documentation from our CLI, it does not affect the CLI that is shipped with the `operator-sdk` and we do not ship this tool with the SDK either.

Reply:
To clarify, given the operator-sdk_add_api.md: https://raw.githubusercontent.com/operator-framework/operator-sdk/master/website/content/en/docs/cli/operator-sdk_add_api.md

How would we replace the `<` and `>` in the *Synopsis* section only? We do not want to change the ones in the *Examples* or any section that is blocked off by the triple tick marks ```

If we were replacing them all I would agree, post processing with a simple sed command would handle the work.

Reply:
IMO: The goal is to solve the format in the docs. I do not care if it will be done during the generation or after. Each field in the doc is related to a cmd field so the solution here shows great. Also, it is just used by the docs and will not change the behaviour of the tool.  Btw shows easy keep maintained so I liked 👍